clvm/run_program.py

```diff
@@ -54,8 +56,12 @@ def run_program(
         pre_eval_op = None
 
     def traverse_path(sexp: SExp, env: SExp) -> Tuple[int, SExp]:
-        cost = PATH_LOOKUP_COST_PER_LEG
-        if sexp.nullp():
+        cost = TRAVERSE_BASE_COST
+        atom = sexp.atom
+        while len(atom) and atom[0] == b"\0":
```

Comment:
I think it would be easier to understand this if the cost was called `TRAVERSE_COST_PER_REDUNDANT_BYTE` or something to indicate that it's a counter-measure.

But wouldn't it be better to just not allow any redundant bytes in the blob? i.e. just hard-fail if there are any bytes left at the end.

Reply:
The current behaviour is to treat the number as an integer, which may actually require the leading 0x00. I don't think we have many tests of deep traversal, so we probably need them to find edge cases and test costs.

Reply:
in the performance benchmark I generate tests of deep lookups, but they aren't testing correctness

Reply:
I agree that regardless of the validity of these paths, we need new tests to cover this

Reply:
> The current behaviour is to treat the number as an integer, which may actually require the leading 0x00.

We only treat it as an integer to the extent that we convert it to a bigint value. iirc, we don't really care whether the value is negative or positive, we just look at one bit at a time, right? It's not clear to me that there *are* any cases were you need to add a 0-prefix. However, I realize the code might look a bit funny having negative paths in it. Although, we could have a convention to represent paths as hex blobs.